cpp/src/sort/is_sorted.cu

```diff
@@ -36,10 +36,17 @@ auto is_sorted(
 {
     cudaStream_t stream = 0;
     auto in_d = table_device_view::create(in);
+    rmm::device_vector<order> d_column_order(column_order);
+    rmm::device_vector<null_order> d_null_precedence;
+    
+    if (has_nulls) {
+        d_null_precedence = null_precedence;
+    }
+
     auto ineq_op = row_lexicographic_comparator<has_nulls>(
         *in_d, *in_d,
-        rmm::device_vector<order>(column_order).data().get(),
-        rmm::device_vector<null_order>(null_precedence).data().get());
+        d_column_order.data().get(),
+        has_nulls ? d_null_precedence.data().get() : nullptr);
```

Comment:
```suggestion
        d_null_precedence.data().get());
```

Reply:
`data().get()` works for default initialized `device_vector`? I tried to look up thrust source for this, but couldn't find it. :disappointed:

Reply:
Yep, you just can't dereference it. 

It would be pretty bad if there was a function like `data` that you weren't allowed to call for some states of an object. Otherwise you couldn't write code like this because if `v` is default constructed you'd have to special case, which would be super annoying. 

```
void foo(device_vector const& v){
   auto p = v.data().get();
}
```

Reply:
Which makes sense, because this is c++ and no-body deleted the default constructor. :D